rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -143,56 +146,73 @@ object Reduce {
       * @param par
       * @return
       */
-    override def eval(par: Par)(implicit env: Env[Par]): M[Unit] = {
+    override def eval(par: Par)(implicit env: Env[Par], rand: Blake2b512Random): M[Unit] = {
+      val filteredExprs = par.exprs.filter { expr =>
+        expr.exprInstance match {
+          case _: EVarBody    => true
+          case _: EEvalBody   => true
+          case _: EMethodBody => true
+          case _              => false
+        }
+      }
+      val starts = Vector(par.receives.size,
```

Comment:
Can you add a comment explaining why do you do that?

Reply:
Why we accumulate the starts vector?